app/src/main/java/com/io/movio/ui/moviesearch/MovieSearchFragment.kt

```diff
@@ -0,0 +1,92 @@
+package com.io.movio.ui.moviesearch
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.Navigation
+import com.io.movio.R
+import com.io.movio.databinding.FragmentMovieSearchBinding
+import com.io.movio.domain.Result
+import com.io.movio.ui.adapter.MoviesAdapter
+import java.time.Year
+import java.util.*
+import kotlin.concurrent.schedule
+
+class MovieSearchFragment : Fragment(), MoviesAdapter.ItemOnClickListener {
+
+    private lateinit var binding: FragmentMovieSearchBinding
+    private val viewModel: MovieSearchViewModel by lazy { ViewModelProvider(this@MovieSearchFragment)[MovieSearchViewModel::class.java] }
+    private var adapter = MoviesAdapter(this)
+    private var timer = Timer()
+    private var queryParam = ""
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+
+        binding = FragmentMovieSearchBinding.inflate(inflater)
+        return binding.root
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.resultList.adapter = adapter
+
+        if (queryParam.isEmpty()) {
+            queryParam = Year.now().toString()
+            viewModel.getMoviesBySearch(queryParam)
```

Comment:
your initial query will always be empty, you can simply make the request in the init block of the view model